app/models/host/discovered.rb

```diff
@@ -45,7 +45,7 @@ def self.import_host_and_facts facts
     hostname = FacterUtils::bootif_mac(facts).try(:downcase).try(:gsub,/:/,'').try(:sub,/^/, hostname_prefix)
 
     # create new host record
-    h = ::Host::Discovered.find_by_name hostname
```

Comment:
I know this isn't yours, but if you're at it, can you change ```h``` into ```host``` ?
I really dislike pointless variable names, and it goes against the Ruby conventions...